### PR TITLE
Add common component

### PR DESCRIPTION
**Pull request by phamilton**
Initial commit of the "common" component of the knative-kafka project as well as top level files (README etc.).  Additional components will be added in separate PRs when the common code has been merged.

### CODE REVIEW

**Review by devguyio — CHANGES_REQUESTED** [review]
Initial check on mandatory files and repo structure as per the [template](https://github.com/kyma-project/community/tree/master/guidelines/repository-guidelines/repository-template) shows two missing files: CODEOWNERS and LICENSE.

File: README.md
Comment (devguyio) [inline README.md]: Maybe remove the reference of `konduit` from the filename and from the diagram?

File: components/common/Makefile
Comment (devguyio) [inline components/common/Makefile]: probably best to change this URL to a gcs repo of Kyma?

File: components/common/pkg/kafka/constants/constants.go
Comment (devguyio) [inline components/common/pkg/kafka/constants/constants.go]: as above, a konduit reference

File: components/common/pkg/prometheus/README.md
Comment (devguyio) [inline components/common/pkg/prometheus/README.md]: Konduit reference

**Review by devguyio — CHANGES_REQUESTED** [review]
Second review round shows some names and URLs that need to change

File: README.md
Comment (devguyio) [reply README.md]: also noticed now `CPI`, `IFLOW` and `C4C`

Comment (phamilton) [reply README.md]: For now I'll just remove the diagram, we can redo it later if it makes sense.

File: components/common/Makefile
Comment (phamilton) [reply components/common/Makefile]: Actually for the common component its no longer needed, was only needed to be compatible with SAP build systems.  So I removed it.

File: components/common/pkg/prometheus/README.md
Comment (phamilton) [reply components/common/pkg/prometheus/README.md]: Good catch, updated.

File: components/common/pkg/kafka/constants/constants.go
Comment (phamilton) [reply components/common/pkg/kafka/constants/constants.go]: Good catch.

**Review by devguyio — APPROVED** [review]
lgtm

**Pull request merged**
